File: guides/micronaut-data-oracle-transaction-priority/java/src/main/java/example/micronaut/Account.java
package example.micronaut;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity("priority_account")
public class Account {

    @Id
    private Long id;

    private String name;

    private Double balance;

    private Status status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getBalance() {
        return balance;
    }

    public void setBalance(Double balance) {
        this.balance = balance;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }
}

File: guides/micronaut-data-oracle-lockfree-reservation/kotlin/src/main/kotlin/example/micronaut/domain/Account.kt
package example.micronaut.domain

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Reservable
import io.micronaut.serde.annotation.Serdeable

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero

@Serdeable
@MappedEntity("ACCOUNT")
data class Account(
    @field:Id
    @field:GeneratedValue
    val id: Long? = null,
    @field:NotBlank
    val name: String,
    @field:Reservable
    @field:PositiveOrZero
    val balance: Long,
    @field:Reservable
    @field:PositiveOrZero
    val credit: Long
)
